index_impl.go

```diff
@@ -830,3 +831,16 @@ func (f *indexImplFieldDict) Close() error {
 	}
 	return f.indexReader.Close()
 }
+
+// helper function to remove duplicate entries from slice of strings
+func deDuplicate(fields []string) []string {
+	entries := make(map[string]bool)
```

Comment:
can use empty struct as value here instead of bool

Reply:
will do.